v2/fake/fake.go

```diff
@@ -89,13 +90,17 @@ type FakeClient struct {
 
 var _ v2.Client = &FakeClient{}
 
+// Actions is a method defined on FakeClient that returns the actions taken on it
 func (c *FakeClient) Actions() []Action {
 	c.Mutex.Lock()
 	defer c.Mutex.Unlock()
 
 	return c.actions
 }
 
+// GetCatalog is a method defined on FakeClient that appends the GetCatalog Action to
```

Comment:
I find the lint errors on implementations of interface methods for doc very annoying.  I think it would be totally sufficient to say "GetCatalog implements the Client.GetCatalog method for the FakeClient."

Your choice

Reply:
they're very annoying. haha 
I think, the less verbose the better, but not having intimate knowledge of the code base caused me to over explain a bit.